src/rules/selector-id-pattern/index.js

```diff
@@ -24,10 +24,8 @@ export default function (pattern) {
     const normalizedPattern = isString(pattern) ? new RegExp(pattern) : pattern
 
     root.walkRules(rule => {
-      // Ignore Sass intepolation possibilities
-      if (/#{.+}/.test(rule.selector)) { return }
-
-      if (cssRuleHasSelectorEndingWithColon(rule)) { return }
+      if (!isStandardRule(rule)) { return }
```

Comment:
I'm not sure Sass interpolation should be prevented in isStandardRule. Because the bulk of the rule might be totally standard. Maybe there should be a isStandardSelector function, instead?